lib/auth/server_info.go

```diff
@@ -70,32 +70,59 @@ func (a *Server) ReconcileServerInfos(ctx context.Context) error {
 	}
 }
 
-func (a *Server) setCloudLabelsOnNodes(ctx context.Context, nodes []types.Server) (failedUpdates int, err error) {
+// getServerInfoNames gets the names of ServerInfos that could exist for a
+// node.
+func getServerInfoNames(node types.Server) []string {
+	var names []string
+	if meta := node.GetCloudMetadata(); meta != nil && meta.AWS != nil {
+		names = append(names, types.ServerInfoNameFromAWS(meta.AWS.AccountID, meta.AWS.InstanceID))
+	}
+	// Manually added ServerInfos should override any other ServerInfos.
```

Comment:
I get the comment, but I don't get it in the context of this method. What manually added infos do we see here?

Reply:
I believe that "manually added ServerInfos" are the node-name-based ones. The AWS ones are created by the discovery process in lib/srv/server/ec2_watcher.go. Not sure where it is best to record this information in a comment - perhaps this could say:

    // Node name based ServerInfos are manually added and override the AWS ServerInfos automatically
    // added by discovery. Labels from ServerInfos later in the list will override the earlier ones.

or something like that perhaps.

Reply:
I've removed the "manually added" language since the matching method is really the only thing that determines the order.